libpromises/parser.c

```diff
@@ -168,6 +168,10 @@ int ParserWarningFromString(const char *warning_str)
     {
         return PARSER_WARNING_ALL;
     }
+    else if (strcmp("truncated", warning_str) == 0)
```

Comment:
`StringEqual()`

Reply:
If i change only this line it will be inconsitent as there will be `strcmp`s above and just one `StringEqual` at the bottom. Should i change the above too?

Reply:
Either change those in a separate commit or leave `strcmp()` here too.